src/middlewared/middlewared/plugins/smb_/util_smbconf.py

```diff
@@ -262,17 +234,6 @@ def generate_smb_conf_dict(
                 else:
                     domain = smb_service_config['workgroup']
             case 'DS_TYPE_LDAP':
-                if not ldap_enabled or not lc['has_samba_schema']:
```

Comment:
Maybe add a comment why this is an empty case.
We _could_ add a log message if something is unexpectedly calling it.